darwin/client.py

```diff
@@ -1,282 +1,58 @@
-            Dictionary which contains the server response
-
-        Raises
-        ------
-        NotFound
-            Resource not found
-        Unauthorized
-            Action is not authorized
-        """
-
-        response: requests.Response = requests.get(urljoin(self.url, endpoint), headers=self._get_headers(team))
-
-        if debug:
-            print(
-                f"Client get request response ({response.json()}) with unexpected status "
-                f"({response.status_code}). "
-                f"Client: ({self})"
-                f"Request: (endpoint={endpoint})"
-            )
-
-        if response.status_code == 401:
-            raise Unauthorized()
-
-        if response.status_code == 404:
-            raise NotFound(urljoin(self.url, endpoint))
-
-        if response.status_code != 200 and retry:
-            time.sleep(10)
-            return self.get(endpoint=endpoint, retry=False)
-
-        if raw:
-            return response
-        else:
-            return self._decode_response(response, debug)
-
-    def put(
-        self,
-        endpoint: str,
-        payload: Dict[str, Any],
-        team: Optional[str] = None,
-        retry: bool = False,
-        debug: bool = False,
-        raw: bool = False,
-    ) -> Union[Dict[str, Any], requests.Response]:
-        """
-        Put something on the server trough HTTP
-
-        Parameters
-        ----------
-        endpoint : str
-            Recipient of the HTTP operation
-        payload : dict
-            What you want to put on the server (typically json encoded)
-        retry : bool
-            Retry to perform the operation. Set to False on recursive calls.
-        debug : bool
-            Debugging flag. In this case failed requests get printed
-        raw : bool
-            Flag for returning raw response
-
-        Returns
-        -------
-        dict
-            Dictionary which contains the server response
-        """
-        response: requests.Response = requests.put(
-            urljoin(self.url, endpoint), json=payload, headers=self._get_headers(team)
-        )
-
-        if debug:
-            print(
-                f"Client GET request got response ({response.json()}) with unexpected status "
-                f"({response.status_code}). "
-                f"Client: ({self})"
-                f"Request: (endpoint={endpoint}, payload={payload})"
-            )
-
-        if response.status_code == 401:
-            raise Unauthorized()
-
-        if response.status_code == 429:
-            error_code = response.json()["errors"]["code"]
-            if error_code == "INSUFFICIENT_REMAINING_STORAGE":
-                raise InsufficientStorage()
-
-        if response.status_code != 200 and retry:
-            time.sleep(10)
-            return self.put(endpoint, payload=payload, retry=False)
-
-        if raw:
-            return response
+        if log is None:
+            self.log: Logger = logging.getLogger()
```

Comment:
better to do `logging.getLogger('darwin')`

Reply:
Done